elasticapm/contrib/celery/__init__.py

```diff
@@ -68,24 +68,30 @@ def set_celery_headers(headers=None, **kwargs):
     transaction = execution_context.get_transaction()
     if transaction is not None:
         trace_parent = transaction.trace_parent
-        trace_parent_string = trace_parent.to_string()
 
-        headers.update({"elasticapm": {"trace_parent_string": trace_parent_string}})
+        # Customize parent span id (if provided)
+        apm_headers = headers.get('elasticapm', dict())
+        if 'parent_span_id' in apm_headers:
+            trace_parent = trace_parent.copy_from()
+            trace_parent.span_id = apm_headers['parent_span_id']
+
+        apm_headers['trace_parent_string'] = trace_parent.to_string()
+        headers.update(elasticapm=apm_headers)
 
 
 def get_trace_parent(celery_task):
     """
     Return a trace parent contained in the request headers of a Celery Task object or None
     """
-    trace_parent = None
-    with suppress(AttributeError, KeyError, TypeError):
-        if celery_task.request.headers is not None:
```

Comment:
This is unsafe because other tracing tools may also set the inner headers to add their tracing ids, whereas we just use the task headers. When using both elasticapm and sentry, elastic's traceparent propagation breaks.